File: 卡码/代码随想录/Go/二叉树/01-二叉树遍历方式/LeetCode_0102.go
//102. 二叉树的层序遍历
package main

import (
	. "binaryTree/types"
)

func levelOrder(root *TreeNode) [][]int {
	v:=make([]int,0)
	res:=make([][]int,0)
	queue:=make([]*TreeNode,0)
	var size int
	if root!=nil {
		queue=append(queue,root)
	}
	for len(queue)!=0 {
		size=len(queue)
		v=[]int{}
		for size>0 {
			node:=queue[0]
			queue=queue[1:]
			v=append(v,node.Val)
			if node.Left!=nil {
				queue=append(queue,node.Left)
			}
			if node.Right!=nil {
				queue=append(queue,node.Right)
			}
			size--
		}
		res=append(res,v)
	}
	return res
}

File: 卡码/代码随想录/Python/二叉树/01-二叉树遍历方式/LeetCode_0102.py
#102. 二叉树的层序遍历
from typing import *
from ..TreeNode import *

class Solution:
  def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
		res:List[List[int]]=[]
		q:List[TreeNode]=[]
		if root!=None:
			q.append(root)
		while len(q)!=0:
			size:int=len(q)
			v:List[int]=[]
			while size>0:
				size-=1
				node:TreeNode=q.pop(0)
				v.append(node.val)
				if node.left:
					q.append(node.left)
				if node.right:
					q.append(node.right)
			res.append(v)
		return res
